### PR TITLE
Bypass NVIDIA Ampere related tuning

### PR DESCRIPTION
**Pull request by wenkaidu**
## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._

**What were the changes?**  
Bypass NVIDIA Ampere related tuning

**Why were the changes made?**  
NVIDIA Ampere code paths get executed for AMD GPUs

**How was the outcome achieved?**  
Bypass NVIDIA Ampere related tuning on AMD GPUs

**Additional Documentation:**  

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

File: src/graph/paths.cc [hunk 1]
```diff
@@ -452,8 +455,11 @@ ncclResult_t ncclTopoNeedFlush(struct ncclTopoSystem* system, int64_t busId, int
   int g;
   NCCLCHECK(ncclTopoIdToIndex(system, GPU, busId, &g));
   struct ncclTopoNode* gpu = system->nodes[GPU].nodes+g;
+#if defined(__HIP_PLATFORM_AMD__) || defined(__HCC__) || defined(__HIPCC__)
+#else
```
Comment (nusislam) [inline src/graph/paths.cc:459]: So, on AMD platform, we will not flush the data because it is not required.

Comment (wenkaidu) [reply src/graph/paths.cc:459]: yes

Comment (nusislam) [reply src/graph/paths.cc:459]: Does it change anything in terms of performance?

Comment (wenkaidu) [reply src/graph/paths.cc:459]: I think flush incurs a few microseconds latency

**Review by nusislam — APPROVED** [review]
LGTM. The PR can be merged after the CI passes.

**Pull request merged**